In [17]:
import pandas as pd
df_customers = pd.read_csv("C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_customers_dataset.csv",dtype={'customer_zip_code_prefix': 'object'})
print("--- 《客户信息表》宏观鸟瞰报告 ---")
df_customers.info()

--- 《客户信息表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  object
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: object(5)
memory usage: 3.8+ MB


In [18]:
print("--- 《客户信息表》ID唯一性审查报告 ---")
unique_customer_count = df_customers['customer_unique_id'].nunique()
print(f"平台总共有 {unique_customer_count} 名独立的客户。")

--- 《客户信息表》ID唯一性审查报告 ---
平台总共有 96096 名独立的客户。


In [19]:
total_orders = len(df_customers)
if total_orders > unique_customer_count:
    repeat_purchase_count = total_orders - unique_customer_count
    print(f"在总共 {total_orders} 笔订单中，我们发现了 {repeat_purchase_count} 次‘复购’行为！")
    print("这意味着，我们，拥有，一批，宝贵的‘忠诚客户’！")
else:
    print("未发现明显的复购行为，平台用户以新客为主。")
print("\n--- 客户地理分布概览 ---")
print("\n客户数量Top 10的州:")
print(df_customers['customer_state'].value_counts().head(10))
print("\n客户数量Top 10的城市:")
print(df_customers['customer_city'].value_counts().head(10))

在总共 99441 笔订单中，我们发现了 3345 次‘复购’行为！
这意味着，我们，拥有，一批，宝贵的‘忠诚客户’！

--- 客户地理分布概览 ---

客户数量Top 10的州:
customer_state
SP    41746
RJ    12852
MG    11635
RS     5466
PR     5045
SC     3637
BA     3380
DF     2140
ES     2033
GO     2020
Name: count, dtype: int64

客户数量Top 10的城市:
customer_city
sao paulo                15540
rio de janeiro            6882
belo horizonte            2773
brasilia                  2131
curitiba                  1521
campinas                  1444
porto alegre              1379
salvador                  1245
guarulhos                 1189
sao bernardo do campo      938
Name: count, dtype: int64


In [21]:
import pandas as pd
df_items = pd.read_csv("C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_order_items_dataset.csv",dtype={'order_item_id': 'object'})
print("--- 《订单商品表》宏观鸟瞰报告 ---")
df_items.info()
print("\n--- 《订单商品表》数值健康体检报告 ---")
print(df_items.describe())
print("\n--- 《订单商品表》ID唯一性审查报告 ---")
primary_key_is_unique = not df_items.duplicated(subset=['order_id', 'order_item_id']).any()
if primary_key_is_unique:
    print("审查通过：‘订单ID + 订单商品序号ID’的组合，是唯一的。表结构健康！")
else:
    print("警告！发现重复的‘订单-商品’记录！数据存在严重质量问题！")

--- 《订单商品表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  object 
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), object(5)
memory usage: 6.0+ MB

--- 《订单商品表》数值健康体检报告 ---
               price  freight_value
count  112650.000000  112650.000000
mean      120.653739      19.990320
std       183.633928      15.806405
min         0.850000       0.000000
25%        39.900000      13.080000
50%        74.990000      16.260000
75%       134.900000      21.150000
max      6735.000000     409.680000

--- 《订单

In [24]:
import pandas as pd
df_payments = pd.read_csv("C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_order_payments_dataset.csv",dtype={'payment_sequential': 'object', 'payment_installments': 'object'})
print("--- 《订单支付表》宏观鸟瞰报告 ---")
df_payments.info()
print("\n--- 《订单支付表》数值健康体检报告 ---")
print(df_payments.describe())
print("\n--- 《订单支付表》业务逻辑探索报告 ---")
print("\n支付类型分布情况:")
print(df_payments['payment_type'].value_counts())
max_sequential = pd.to_numeric(df_payments['payment_sequential']).max()
if max_sequential > 1:
    print(f"\n发现‘混合支付’行为！一个订单，最多，采用了 {max_sequential} 种支付方式！")
else:
    print("\n未发现‘混合支付’行为。")
credit_card_payments = df_payments[df_payments['payment_type'] == 'credit_card']
print("\n信用卡支付的分期情况:")
print(credit_card_payments['payment_installments'].value_counts(normalize=True).head(5))

--- 《订单支付表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  object 
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  object 
 4   payment_value         103886 non-null  float64
dtypes: float64(1), object(4)
memory usage: 4.0+ MB

--- 《订单支付表》数值健康体检报告 ---
       payment_value
count  103886.000000
mean      154.100380
std       217.494064
min         0.000000
25%        56.790000
50%       100.000000
75%       171.837500
max     13664.080000

--- 《订单支付表》业务逻辑探索报告 ---

支付类型分布情况:
payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

发现‘混合支付’行为！一个订单，最多，采用了 29 种支付方式！

信用卡支付的分期情况:
payment_ins

In [27]:
import pandas as pd
df_reviews = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_order_reviews_dataset.csv",
    parse_dates=['review_creation_date', 'review_answer_timestamp'])   
print("--- 《订单评价表》宏观鸟瞰报告 ---")
df_reviews.info()
print("\n--- 《订单评价表》数值健康体检报告 ---")
print(df_reviews['review_score'].describe())
print("\n--- 《订单评价表》业务逻辑探索报告 ---")
print("\n评价分数分布情况 (数量):")
print(df_reviews['review_score'].value_counts())
print("\n评价分数分布情况 (百分比):")
print(df_reviews['review_score'].value_counts(normalize=True))

--- 《订单评价表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   review_id                99224 non-null  object        
 1   order_id                 99224 non-null  object        
 2   review_score             99224 non-null  int64         
 3   review_comment_title     11568 non-null  object        
 4   review_comment_message   40977 non-null  object        
 5   review_creation_date     99224 non-null  datetime64[ns]
 6   review_answer_timestamp  99224 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(1), object(4)
memory usage: 5.3+ MB

--- 《订单评价表》数值健康体检报告 ---
count    99224.000000
mean         4.086421
std          1.347579
min          1.000000
25%          4.000000
50%          5.000000
75%          5.000000
max          5.000000
Name: review_score, dtype: float64

--- 《订单评价表》业务逻辑

In [28]:
df_reviews['response_time'] = (df_reviews['review_answer_timestamp'] - df_reviews['review_creation_date'])
print("\n平台对用户评价的响应速度分析:")
print(df_reviews['response_time'].describe())


平台对用户评价的响应速度分析:
count                        99224
mean     3 days 03:34:33.029700475
std      9 days 21:21:40.258026234
min                0 days 02:08:29
25%         1 days 00:07:00.750000
50%         1 days 16:11:55.500000
75%                3 days 02:29:08
max              518 days 16:46:52
Name: response_time, dtype: object


In [31]:
import pandas as pd
df_orders = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_orders_dataset.csv",
    parse_dates=[
        'order_purchase_timestamp',
        'order_approved_at',
        'order_delivered_carrier_date',
        'order_delivered_customer_date',
        'order_estimated_delivery_date'])
print("--- 《订单总表》宏观鸟瞰报告 ---")
df_orders.info()
print("\n--- 《订单总表》数值健康体检报告 ---")
print(df_orders.describe(include='all'))
print("\n--- 《订单总表》业务逻辑探索报告 ---")
primary_key_is_unique = df_orders['order_id'].nunique() == len(df_orders)
if primary_key_is_unique:
    print("审查通过：‘order_id’是唯一的，符合主键定义。表结构健康！")
else:
    print("警告！发现重复的‘order_id’！数据存在严重结构性问题！")
print("\n订单状态分布情况:")
print(df_orders['order_status'].value_counts())
    

--- 《订单总表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   order_id                       99441 non-null  object        
 1   customer_id                    99441 non-null  object        
 2   order_status                   99441 non-null  object        
 3   order_purchase_timestamp       99441 non-null  datetime64[ns]
 4   order_approved_at              99281 non-null  datetime64[ns]
 5   order_delivered_carrier_date   97658 non-null  datetime64[ns]
 6   order_delivered_customer_date  96476 non-null  datetime64[ns]
 7   order_estimated_delivery_date  99441 non-null  datetime64[ns]
dtypes: datetime64[ns](5), object(3)
memory usage: 6.1+ MB

--- 《订单总表》数值健康体检报告 ---
                                order_id                       customer_id  \
count                              99441

In [32]:
import pandas as pd
df_orders = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_orders_dataset.csv",
    parse_dates=[
        'order_purchase_timestamp',
        'order_approved_at',
        'order_delivered_carrier_date',
        'order_delivered_customer_date',
        'order_estimated_delivery_date'])
condition_1 = df_orders['order_status'] == 'delivered'
condition_2 = df_orders['order_delivered_customer_date'].isnull()
ghost_orders = df_orders[condition_1 & condition_2]
print("--- “幽灵订单”侦察报告 ---")
if not ghost_orders.empty:
    print(f"成功捕获 {len(ghost_orders)} 个‘幽灵订单’！")
    print("它们的详细情报如下：")
    print(ghost_orders)
else: 
     print("未发现任何‘幽灵订单’。数据完全一致。")
        
        

--- “幽灵订单”侦察报告 ---
成功捕获 8 个‘幽灵订单’！
它们的详细情报如下：
                               order_id                       customer_id  \
3002   2d1e2d5bf4dc7227b3bfebb81328c15f  ec05a6d8558c6455f0cbbd8a420ad34f   
20618  f5dd62b788049ad9fc0526e3ad11a097  5e89028e024b381dc84a13a3570decb4   
43834  2ebdfc4f15f23b91474edf87475f108e  29f0540231702fda0cfdee0a310f11aa   
79263  e69f75a717d64fc5ecdfae42b2e8e086  cfda40ca8dd0a5d486a9635b611b398a   
82868  0d3268bad9b086af767785e3f0fc0133  4f1d63d35fb7c8999853b2699f5c7649   
92643  2d858f451373b04fb5c984a1cc2defaf  e08caf668d499a6d643dafd7c5cc498a   
97647  ab7c89dc1bf4a1ead9d6ec1ec8968a84  dd1b84a7286eb4524d52af4256c0ba24   
98038  20edc82cf5400ce95e1afacc25798b31  28c37425f1127d887d7337f284080a0f   

      order_status order_purchase_timestamp   order_approved_at  \
3002     delivered      2017-11-28 17:44:07 2017-11-28 17:56:40   
20618    delivered      2018-06-20 06:58:43 2018-06-20 07:19:05   
43834    delivered      2018-07-01 17:05:11 2018-07-01 17:1

In [33]:
import pandas as pd
df_orders = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_orders_dataset.csv",
    parse_dates=[
        'order_purchase_timestamp',
        'order_approved_at',
        'order_delivered_carrier_date',
        'order_delivered_customer_date',
        'order_estimated_delivery_date'])
contradiction_condition = (df_orders['order_status'] != 'delivered') & (df_orders[
    'order_delivered_customer_date'].notnull())
contradiction_orders = df_orders[contradiction_condition]
print("--- “矛盾订单”侦察报告 ---")
if not contradiction_orders.empty:
    print(f"成功捕获 {len(contradiction_orders)} 个‘状态与行为’相矛盾的‘异常订单’！")
    print("它们的详细情报如下：")
    print(contradiction_orders)
else:
    print("未发现任何‘矛盾订单’。")

--- “矛盾订单”侦察报告 ---
成功捕获 6 个‘状态与行为’相矛盾的‘异常订单’！
它们的详细情报如下：
                               order_id                       customer_id  \
2921   1950d777989f6a877539f53795b4c3c3  1bccb206de9f0f25adc6871a1bcf77b2   
8791   dabf2b0e35b423f94618bf965fcb7514  5cdec0bb8cbdf53ffc8fdc212cd247c6   
58266  770d331c84e5b214bd9dc70a10b829d0  6c57e6119369185e575b36712766b0ef   
59332  8beb59392e21af5eb9547ae1a9938d06  bf609b5741f71697f65ce3852c5d2623   
92636  65d1e226dfaeb8cdc42f665422522d14  70fc57eeae292675927697fe03ad3ff5   
94399  2c45c33d2f9cb8ff8b1c86cc28c11c30  de4caa97afa80c8eeac2ff4c8da5b72e   

      order_status order_purchase_timestamp   order_approved_at  \
2921      canceled      2018-02-19 19:48:52 2018-02-19 20:56:05   
8791      canceled      2016-10-09 00:56:52 2016-10-09 13:36:58   
58266     canceled      2016-10-07 14:52:30 2016-10-07 15:07:10   
59332     canceled      2016-10-08 20:17:50 2016-10-09 14:34:30   
92636     canceled      2016-10-03 21:01:41 2016-10-04 10:18:57   
9

In [34]:
import pandas as pd
df_products = pd.read_csv("C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_products_dataset.csv")
print("--- 《商品信息表》宏观鸟瞰报告 ---")
df_products.info()
print("\n--- 《商品信息表》数值健康体检报告 ---")
print(df_products.describe())
print("\n--- 《商品信息表》业务逻辑探索报告 ---")
primary_key_is_unique = df_products['product_id'].nunique() == len(df_products)
if primary_key_is_unique:
    print("审查通过：‘product_id’是唯一的，符合主键定义。表结构健康！")
else:
    print("警告！发现重复的‘product_id’！数据存在严重结构性问题！")
print("\n商品品类分布Top 10:")
print(df_products['product_category_name'].value_counts().head(10))

--- 《商品信息表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB

--- 《商品信息表》数值健康体检报告 ---
       product_name_lenght  product_description_lenght  product_photos_qty  \
count         32341.000000                32341.000000        32341.000000   
mean      

In [35]:
import pandas as pd
df_sellers = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_sellers_dataset.csv",
    dtype={'seller_zip_code_prefix': 'object'})
print("--- 《卖家信息表》宏观鸟瞰报告 ---")
df_sellers.info()
print("\n--- 《卖家信息表》数值健康体检报告 ---")    
print(df_sellers.describe())
print("\n--- 《卖家信息表》业务逻辑探索报告 ---")
primary_key_is_unique = df_sellers['seller_id'].nunique() == len(df_sellers)
if primary_key_is_unique:
    print("审查通过：‘seller_id’是唯一的，符合主键定义。表结构健康！")
else:
    print("警告！发现重复的‘seller_id’！数据存在严重结构性问题！")
print("\n卖家数量Top 10的州:")
print(df_sellers['seller_state'].value_counts().head(10))
print("\n卖家数量Top 10的城市:")
print(df_sellers['seller_city'].value_counts().head(10))  

--- 《卖家信息表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   object
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: object(4)
memory usage: 96.8+ KB

--- 《卖家信息表》数值健康体检报告 ---
                               seller_id seller_zip_code_prefix seller_city  \
count                               3095                   3095        3095   
unique                              3095                   2246         611   
top     9e25199f6ef7e7c347120ff175652c3b                  14940   sao paulo   
freq                                   1                     49         694   

       seller_state  
count          3095  
unique           23  
top              SP  
freq      

In [36]:
import pandas as pd
df_geolocation = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_geolocation_dataset.csv",
    dtype={'geolocation_zip_code_prefix': 'object'})
print("--- 《地理位置表》宏观鸟瞰报告 ---")
df_geolocation.info()
print("\n--- 《地理位置表》数值健康体检报告 ---")
print(df_geolocation.describe())

--- 《地理位置表》宏观鸟瞰报告 ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  object 
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  object 
 4   geolocation_state            1000163 non-null  object 
dtypes: float64(2), object(3)
memory usage: 38.2+ MB

--- 《地理位置表》数值健康体检报告 ---
       geolocation_lat  geolocation_lng
count     1.000163e+06     1.000163e+06
mean     -2.117615e+01    -4.639054e+01
std       5.715866e+00     4.269748e+00
min      -3.660537e+01    -1.014668e+02
25%      -2.360355e+01    -4.857317e+01
50%      -2.291938e+01    -4.663788e+01
75%      -1.997962e+01    -4.376771e+01
max       4.506593e+01     1.211054e+02


核心洞察：我们的帝国，是一个，以“圣保罗(SP)”为“绝对中心”，但，充满了“地域性商业活力”与“数据不一致性”的、复
杂而真实的“商业生态”
1. “忠诚”与“陌生”并存的“客户结构”：
   数据事实：我们拥有96,096名独立客户，他们，在两年内，总共，贡献了99,441笔订单。
   商业洞察：这，揭示了一个“健康的、以新客为主，但，已，成功，培养出第一批‘核心粉丝’”的“平台生命周期”！
   那3,345次“复购行为”，就是，我们，未来，可以，去，深入挖掘“用户忠诚度”的“第一座金矿”！
2. “高度聚集”的“卖家生态”：
   数据事实：3,095位卖家，却，仅仅，来自于2,246个“不重复”的“邮政编码”。其中，一个，名为14940的“邮编”，就，聚集了，高达49位“卖家”！
   商业洞察：这，绝不是“随机分布”！这，强烈地，暗示着，我们平台，存在着，基于“地理位置”的“产业集群”！“14
   940”这个地区，很可能，是，某个“特定品类”（比如，纺织品、鞋类）的“生产与批发中心”！这，为我们，未来，
   去，分析“供应链优势”与“区域性竞争”，提供了，最宝贵的“切入点”！
3. “充满陷阱”的“地理坐标”：
   数据事实：我们，用“代码”，无可辩驳地，证明了，同一个`geolocation_zip_code_prefix`，会，对应，多个，不同的`lat/lng`坐标！
   战略结论：这，是，我们“数据清洗”阶段的“最高优先级任务”！我们，必须，通过“取平均值”等方式，为，每一个“
   邮编”，都，创造出一个“唯一”的“官方坐标”，否则，我们，所有的“距离计算”，都将，建立在“流沙”之上！

核心洞察：我们的帝国，其“交易流程”，充满了“多样性”与“潜在的效率瓶颈”。
1. “多步骤、长周期”的“订单生命”：
   数据事实：我们，拥有，从purchase（下单）到approved（支付），再到carrier（发货），最后到customer（送达）的、完整的“时间戳链条”。
   商业洞察：这，就是，我们，预测“履约总时长”这个“核心目标Y”的“数据基石”！同时，它，也，暴露了，我们，整
   个“商业流程”中，所有，可能的“效率瓶颈”！“支付确认”，到底，花了多久？“卖家备货”，又花了多久？“物流运
   输”，再花了多久？——这就是我们未来要进行“归因分析”的“主战场”！
2. “复杂”的“支付行为”：
   数据事实：payment_sequential的最大值，高达29！
   商业洞察：我们的“客户”，是“精打细算”的！他们，会，熟练地，使用“多种支付方式”（比如，信用卡+多张代金券
   ）的“组合拳”，来，完成“一次”购物！这，为我们，去，理解“用户支付习惯”和“促销活动效果”，提供了，最“真实
   ”的“样本”！

核心洞察：我们的帝国，其“商品”与“服务”，正在，接受，一场，充满了“沉默的赞美”与“尖锐的抱怨”的“全民公投”。
1. “信息不完整”的“商品军火库”：
   数据事实：存在610个，几乎“没有任何描述性信息”的“幽灵商品”
   战略结论：这些，是“数据的噪音”。我们，将在“清洗”阶段，对它们，进行“果断”的“清除”，以，确保，我们“模型 ”的“纯净性”。
2. “冰火两重天”的“用户评价”：
   数据事实：超过57%的用户，直接，给出了“5星”满分。但，同时，也有，超过11%的用户，给出了“1星”的“极端差评
   ”。而，只有，不到一半的用户，留下了“文字评论”。
   商业洞察：这，是一个，典型的“U型/J型分布”！它，告诉我们，我们，不能，简单地，去看“平均分”。我们，必须
   ，将“分析的焦点”，同时，放在“是什么，让，那57%的用户，如此‘满意’？”和“是什么，让，那11%的用户，如此‘
   愤怒’？”这两个“极端”的问题上！而，那“4万条”宝贵的“文字评论”，就是，我们，回答这两个问题的、唯一的“钥
   匙”！
3. “跨越语言”的“全球化野心”：
   数据事实：我们，拥有一张，完整的“葡语-英语”的“翻译表”。
   战略结论：这，为我们，未来，将我们的“分析报告”，呈现给“任何一个国家”的“面试官”或“决策者”，扫清了，最后的“语言障碍”。
   

In [37]:
import pandas as pd
df_orders = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_orders_dataset.csv",
    parse_dates=[
        'order_purchase_timestamp',
        'order_approved_at',
        'order_delivered_carrier_date',
        'order_delivered_customer_date',
        'order_estimated_delivery_date'])
df_payments = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_order_payments_dataset.csv",
    dtype={'payment_sequential': 'object', 'payment_installments': 'object'})
df_order_payments = pd.merge(
    left=df_orders,
    right=df_payments,
    on='order_id',
    how='left')
print("--- “订单-支付”连接后，宏观鸟瞰报告 ---")
print(f"连接前的订单总表行数: {len(df_orders)}")
print(f"连接后的宽表行数: {len(df_order_payments)}")
print("\n连接后，数据预览:")
print(df_order_payments.head())
print("\n连接后，数据信息:")
df_order_payments.info()

--- “订单-支付”连接后，宏观鸟瞰报告 ---
连接前的订单总表行数: 99441
连接后的宽表行数: 103887

连接后，数据预览:
                           order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
4  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   

  order_status order_purchase_timestamp   order_approved_at  \
0    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
3    delivered      2018-07-24 20:41:37 2018-07-26 03:24:27   
4    delivered      2018-08-08 08:38:49 2018-08-08 08:55:23   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-

In [38]:
import pandas as pd
df_items = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_order_items_dataset.csv",
    dtype={'order_item_id': 'object'})
df_order_items_payments = pd.merge(
    left=df_order_payments,
    right=df_items,
    on='order_id',
    how='left')
print("--- “订单-支付-商品”连接后，宏观鸟瞰报告 ---")
print(f"连接前的宽表行数: {len(df_order_payments)}")
print(f"连接后的超级宽表行数: {len(df_order_items_payments)}")
print("\n连接后，数据预览:")
print(df_order_items_payments.head())
print("\n连接后，数据信息:")
df_order_items_payments.info()
    
    

--- “订单-支付-商品”连接后，宏观鸟瞰报告 ---
连接前的宽表行数: 103887
连接后的超级宽表行数: 118434

连接后，数据预览:
                           order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
4  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   

  order_status order_purchase_timestamp   order_approved_at  \
0    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
3    delivered      2018-07-24 20:41:37 2018-07-26 03:24:27   
4    delivered      2018-08-08 08:38:49 2018-08-08 08:55:23   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2

In [39]:
import pandas as pd
df_customers = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_customers_dataset.csv",
    dtype={'customer_zip_code_prefix': 'object'})
df_final_temp = pd.merge(
    left=df_order_items_payments,
    right=df_customers,
    on='customer_id',
    how='left')
print("--- “订单-商品-支付-客户”连接后，宏观鸟瞰报告 ---")
print(f"连接前的超级宽表行数: {len(df_order_items_payments)}")
print(f"连接后的究极宽表行数: {len(df_final_temp)}")
print("\n连接后，数据预览:")
print(df_final_temp.head())
print("\n连接后，数据信息:")
df_final_temp.info()


--- “订单-商品-支付-客户”连接后，宏观鸟瞰报告 ---
连接前的超级宽表行数: 118434
连接后的究极宽表行数: 118434

连接后，数据预览:
                           order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
4  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   

  order_status order_purchase_timestamp   order_approved_at  \
0    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
3    delivered      2018-07-24 20:41:37 2018-07-26 03:24:27   
4    delivered      2018-08-08 08:38:49 2018-08-08 08:55:23   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00       

In [40]:
import pandas as pd
df_products = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_products_dataset.csv")
df_sellers = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_sellers_dataset.csv",
    dtype={'seller_zip_code_prefix': 'object'})
df_with_products = pd.merge(
    left=df_final_temp,
    right=df_products,
    on='product_id',
    how='left')
df_with_sellers = pd.merge(
    left=df_with_products,
    right=df_sellers,
    on='seller_id',
    how='left')
print("--- “订单-商品-支付-客户-产品-卖家”连接后，宏观鸟瞰报告 ---")
print(f"连接前的究极宽表行数: {len(df_final_temp)}")
print(f"连接后的最终宽表行数: {len(df_with_sellers)}")
print("\n连接后，数据预览:")
print(f"最终宽表维度: {df_with_sellers.shape}")
print(df_with_sellers.head())
print("\n连接后，数据信息:")
df_with_sellers.info()

--- “订单-商品-支付-客户-产品-卖家”连接后，宏观鸟瞰报告 ---
连接前的究极宽表行数: 118434
连接后的最终宽表行数: 118434

连接后，数据预览:
最终宽表维度: (118434, 33)
                           order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
4  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   

  order_status order_purchase_timestamp   order_approved_at  \
0    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
3    delivered      2018-07-24 20:41:37 2018-07-26 03:24:27   
4    delivered      2018-08-08 08:38:49 2018-08-08 08:55:23   

  order_delivered_carrier_date order_delivered_customer_date  \
0         

In [41]:
import pandas as pd
df_translation = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\product_category_name_translation.csv")
df_geolocation_raw = pd.read_csv(
    "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_geolocation_dataset.csv",
    dtype={'geolocation_zip_code_prefix': 'object'})
df_geolocation = df_geolocation_raw.groupby('geolocation_zip_code_prefix').agg({
    'geolocation_lat': 'mean',
    'geolocation_lng': 'mean'}).reset_index()
df_with_translation = pd.merge(
    left=df_with_sellers,
    right=df_translation,
    on='product_category_name',
    how='left')
df_with_customer_geo = pd.merge(
    left=df_with_translation,
    right=df_geolocation,
    left_on='customer_zip_code_prefix',
    right_on='geolocation_zip_code_prefix',
    how='left')
df_final = pd.merge(
    left=df_with_customer_geo,
    right=df_geolocation,
    left_on='seller_zip_code_prefix',
    right_on='geolocation_zip_code_prefix',
    how='left',
    suffixes=('_customer', '_seller'))
print("--- “最终分析宽表”构建完毕！宏观鸟瞰报告 ---")
print(f"最终宽表的维度: {df_final.shape}")
print("\n最终宽表，数据信息:")
print(df_final[['product_category_name_english', 'geolocation_lat_customer',
                'geolocation_lng_customer', 'geolocation_lat_seller', 'geolocation_lng_seller']].info())
                

--- “最终分析宽表”构建完毕！宏观鸟瞰报告 ---
最终宽表的维度: (118434, 40)

最终宽表，数据信息:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118434 entries, 0 to 118433
Data columns (total 5 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   product_category_name_english  115881 non-null  object 
 1   geolocation_lat_customer       118117 non-null  float64
 2   geolocation_lng_customer       118117 non-null  float64
 3   geolocation_lat_seller         117343 non-null  float64
 4   geolocation_lng_seller         117343 non-null  float64
dtypes: float64(4), object(1)
memory usage: 4.5+ MB
None


In [42]:
final_output_path = "C:\\Users\\Administrator\\Desktop\\OLIST!\\olist_final_analysis_dataset.csv"
df_final.to_csv(final_output_path, index=False)
print(f"--- “数据泰坦”已成功实体化！---")
print(f"我们最宝贵的核心资产，已安全地，存储于：")
print(final_output_path)

--- “数据泰坦”已成功实体化！---
我们最宝贵的核心资产，已安全地，存储于：
C:\Users\Administrator\Desktop\OLIST!\olist_final_analysis_dataset.csv
